### DISCUSSION TITLE
How to convert a query into a hashmap?

### DISCUSSION THREAD

**Discussion by oriongonza** [rusqlite/rusqlite#1261] [Questions]
I'm not sure if this is the place to ask but... 

I've looked around the docs for a while, couldn't find much info and I ended up doing it like this. It doesn't look good and I'm sure there's a much better way:
```rust
fn main() {
    let con = Connection::open("./data.db").unwrap();
    let mut imgs: HashMap<i32, String> = HashMap::new();
    let mut x = con.prepare("select * from imgs").unwrap();
    x.query_map([], |x| {
        Ok((x.get::<usize, i32>(0), x.get::<usize, String>(1)))
    })
    .into_iter()
    .flatten()
    .flatten()
    .for_each(|x| {imgs.insert(x.0.unwrap(), x.1.unwrap());});
}

```

**Comment by gwenn** [comment]
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=01199c00f1e88ba061c5923bf6891578
Or with `FallibleIterator`:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=705995c5bfd3de13c265aff9229db698

**Reply by oriongonza** [reply]
Good to know, I appreciate it. 
It might be helpful to add this as some sort of helper function inside of the library (the primary key as the key and a tuple of values for the rows, for example).
